### PR TITLE
Marshaller Flyout - POPUP

### PR DESCRIPTION
**Pull request by mlzummo**

### CODE REVIEW

**Comment by mlzummo** [comment]
@GordonSmith not sure how i am going to overcome figuring out how i can get the popup to autosize

**Comment by mlzummo** [comment]
@GordonSmith Also should the popup be movable?

**Comment by mlzummo** [comment]
http://rawgit.com/mlzummo/Visualization/MARSHALLER_FLYOUT_2/demos/marshaller/index.html?Hostname=10.173.147.1&Port=8010&Protocol=http&Wuid=W20151102-091253&ResultName=leeddx_issue_986_formddl_Comp_Ins002_DDL&HTML

File: src/marshaller/HTML.js [hunk 1]
```diff
@@ -110,12 +134,81 @@
...
+                                        label: "\uf00d",
+                                        width: 20,
+                                        padding: "0px 5px",
+                                        class: "close",
+                                        font: "FontAwesome",
+                                    }
+                                ])
+                                .on("click", function(obj) {
+                                    if (obj.class === "close") {
+                                        popup
```
Comment (mlzummo) [inline src/marshaller/HTML.js]: might need to set visible to false after the render?

File: src/marshaller/HTML.js [hunk 2]
```diff
@@ -1,15 +1,17 @@
...
+        root.marshaller_HTML = factory(root.d3, root.layout_Grid, root.marshaller_HipieDDL, root.layout_Surface, root.layout_Cell, root.layout_Popup);
     }
-}(this, function (d3, Grid, HipieDDL, Surface, Cell) {
+}(this, function (d3, Grid, HipieDDL, Surface, Cell, Popup) {
     function HTML() {
         Grid.call(this);
 
         this.surfacePadding(0);
+
+        this.flyoutWidgets = {};
```
Comment (mlzummo) [inline src/marshaller/HTML.js]: should i use a widgetarray ... i dont think this stuff should be serialized though ....

File: src/marshaller/HTML.js [hunk 3]
```diff
@@ -110,12 +134,81 @@
                         viz.widgetSurface.title(viz.title);
                         context.setContent(cellRow, cellCol, viz.widgetSurface);
                         vizCellMap[viz.id] = context.getWidgetCell(viz.widgetSurface.id());
+
                         cellCol++;
                     });
+
+                    popupVisualizations.forEach(function (viz, idx) {
+                        var popup = context.flyoutWidgets[viz.id] = new Popup()
+                            //.size({width:400, height:400}) // need to figure out this
```
Comment (mlzummo) [inline src/marshaller/HTML.js]: can't figure out how to autosize ...

**Comment by mlzummo** [comment]
GH-745

File: src/marshaller/HTML.js [hunk 4]
```diff
@@ -110,12 +134,81 @@
...
+                                        padding: "0px 5px",
+                                        class: "popup-flyout",
+                                        font: "FontAwesome",
+                                       };
+                            });
+                            vizCellMap[viz.id].widget()
+                                .buttonAnnotations(buttonAnnotations)
+                                .on("click", function(obj) {
+                                    if (obj.class === "popup-flyout") {
+                                        var idx = obj.id.split("_").slice(-1).pop();
```
Comment (mlzummo) [inline src/marshaller/HTML.js]: prolly dont need slice

**Comment by GordonSmith** [comment]
1.  Having the popup autosize should be the job of the popup layout widget
2.  No need to have the popup widget be movable (again that would be the job of the popup layout widget).

**Comment by mlzummo** [comment]
@GordonSmith can you review this and respond to my two questions above? going to be basing some other code off this  want to make sure this is going to get in there or not

**Comment by GordonSmith** [comment]
The location of the flyout logic looks wrong to me - if you look at the first "flyout" visualization in the DDL:
```
         {
            "id":"Filterpie",
            "source":{

            },
            "type":"FORM",
            "title":"Filtertable1",
            "fields":[
               {
                  "id":"productline",
                  "properties":{
                     "default":"Ciaz",
                     "label":"productline",
                     "type":"string"
                  }
               }
            ],
            "properties":{
               "flyout":true
            },
            "events":{
               "click":{
                  "mappings":{
                     "productline":"productline"
                  },
                  "updates":[
                     {
                        "visualization":"pietest",
                        "instance":"Ins002",
                        "datasource":"Ins002_dsOutput",
                        "merge":false
                     }
                  ]
               }
            }
         },
```
What I would expect to happen is:
1. Create this visualization (FORM) as normal - but don't let it get added to the HTML/Graph/Tabbed Marshallers Grid/Graph layouts.
2. Create the pietest visualization as normal
2. Add a PopupWidget button to the pietest surface title (with this FormWidget as the content)
3. When clicked this button will open a popup widget with this visualization (FORM)

So now this Issue becomes 2 different chunks of work.
1.  Check that Surface + PopupButton (containing a FormWidget) all works.
2.  Tweak the HipieDDL to initialize the above correctly (I can do this).

**Comment by mlzummo** [comment]
@GordonSmith going to ignore above comment, waiting for new ones we talked about today

**Pull request merged**